lib/pure/random.nim

```diff
@@ -669,6 +715,20 @@ when isMainModule:
     doAssert a[0] == 1
     doAssert a[1] == 0
 
+    const b = @[0, 1]
+    let c = shuffled(b)
+    doAssert b[0] == 0
+    doAssert b[1] == 1
+    doAssert c[0] == 1
+    doAssert c[1] == 0
+
+    const d = @[0, 1]
+    let e = shuffled(b)
+    doAssert d[0] == 0
+    doAssert d[1] == 1
+    doAssert e[0] == 1
+    doAssert e[1] == 0
+
```

Comment:
you wrote 4 function definitons, but only tested 2 of them. And those tests only barely test the functionality.

Reply:
I have combined the four procedures into one procedure.